Sensor integration staging

sohampatkar10 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184)
<!-- Reviewable:end -->

garimellagowtham commented:
Reviewed 11 of 11 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[include/aerial_autonomy/sensors/base_sensor.h, line 29 at r1](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184#-LCoO_D_BCI48IhhIrsJ:-LCoO_D_BCI48IhhIrsK:b9lg22w) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/d9e5b082ceac66c040478fd2d5065528e0bc8004/include/aerial_autonomy/sensors/base_sensor.h#L29)):*
> ```objc
> * and status
> */
> template <class SensorDataT> class Sensor {
> ```

Should the sensor type match the one in controller connector or all the data provided by the sensor?

---

*[include/aerial_autonomy/sensors/velocity_sensor.h, line 22 at r1](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184#-LCoJe3b0H_pZBhOFT8I:-LCoJe3b0H_pZBhOFT8J:bwprn8i) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/d9e5b082ceac66c040478fd2d5065528e0bc8004/include/aerial_autonomy/sensors/velocity_sensor.h#L22)):*
> ```objc
>   VelocitySensor(VelocitySensorConfig config) : config_(config) {
>     VLOG(2) << "Initialzing ROS Sensor";
>     odom_sub_ = nh_.subscribe("/velocity_sensor/odometry", 1,
> ```

The topic should be part of the config

---

*[include/aerial_autonomy/sensors/velocity_sensor.h, line 76 at r1](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184#-LCoKByyDsuXI7nqn7--:-LCoKByyDsuXI7nqn7-0:b-3osm2w) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/d9e5b082ceac66c040478fd2d5065528e0bc8004/include/aerial_autonomy/sensors/velocity_sensor.h#L76)):*
> ```objc
>   ros::Subscriber odom_sub_;
>   /**
>   * @brief sensor's origin in world frame
> ```

Is sensor's origin fixed in world frame? For Guidance, the velocity is in body velocity?

---

*[proto/velocity_sensor_config.proto, line 8 at r1](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184#-LCoN8yi28gx4eHlSsDi:-LCoN8yi28gx4eHlSsDj:bm0jjbn) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/d9e5b082ceac66c040478fd2d5065528e0bc8004/proto/velocity_sensor_config.proto#L8)):*
> ```protobuf
>   * @brief transform of sensor in robot's frame
>   */
>   optional config.Transform sensor_transform = 1;
> ```

In robots frame?

---


*Comments from [Reviewable](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184)*
<!-- Sent from Reviewable.io -->

sohampatkar10 commented:
Review status: 8 of 11 files reviewed at latest revision, 4 unresolved discussions.

---

*[include/aerial_autonomy/sensors/base_sensor.h, line 29 at r1](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184#-LCoO_D_BCI48IhhIrsJ:-LCohaW4C2bbnWBcXEXx:bjubh8v) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/d9e5b082ceac66c040478fd2d5065528e0bc8004/include/aerial_autonomy/sensors/base_sensor.h#L29)):*
<details><summary><i>Previously, garimellagowtham (Gowtham Garimella) wrote…</i></summary><blockquote>

Should the sensor type match the one in controller connector or all the data provided by the sensor?
</blockquote></details>

Keeping as is for now

---

*[include/aerial_autonomy/sensors/velocity_sensor.h, line 22 at r1](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184#-LCoJe3b0H_pZBhOFT8I:-LCohRRa7nKPDeqO5zvO:b-896fix) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/d9e5b082ceac66c040478fd2d5065528e0bc8004/include/aerial_autonomy/sensors/velocity_sensor.h#L22)):*
<details><summary><i>Previously, garimellagowtham (Gowtham Garimella) wrote…</i></summary><blockquote>

The topic should be part of the config
</blockquote></details>

Done.

---

*[include/aerial_autonomy/sensors/velocity_sensor.h, line 76 at r1](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184#-LCoKByyDsuXI7nqn7--:-LCohTs0B0CwXx-RYj5g:b-huej89) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/d9e5b082ceac66c040478fd2d5065528e0bc8004/include/aerial_autonomy/sensors/velocity_sensor.h#L76)):*
<details><summary><i>Previously, garimellagowtham (Gowtham Garimella) wrote…</i></summary><blockquote>

Is sensor's origin fixed in world frame? For Guidance, the velocity is in body velocity?
</blockquote></details>

Removed transform

---

*[proto/velocity_sensor_config.proto, line 8 at r1](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184#-LCoN8yi28gx4eHlSsDi:-LCohY840Sp2zfIsnQcn:b-huej89) ([raw file](https://github.com/jhu-asco/aerial_autonomy/blob/d9e5b082ceac66c040478fd2d5065528e0bc8004/proto/velocity_sensor_config.proto#L8)):*
<details><summary><i>Previously, garimellagowtham (Gowtham Garimella) wrote…</i></summary><blockquote>

In robots frame?
</blockquote></details>

Removed transform

---


*Comments from [Reviewable](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184)*
<!-- Sent from Reviewable.io -->

msheckells commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/jhu-asco/aerial_autonomy/184)*
<!-- Sent from Reviewable.io -->
